src/bom/bitcode.rs

```diff
@@ -283,53 +282,36 @@ fn build_bitcode_arguments(chan : &mut mpsc::Sender<Option<Event>>,
 
     // Next, copy over all of the flags we want to keep
     let mut it = orig_args.iter();
-    let mut skip_next = false;
-    let mut next_is_val = false;
     while let Some(arg) = it.next() {
-        // Skip value argument to a previous blacklisted argument
-        if skip_next {
-            skip_next = false;
-            continue;
-        }
-
-        if next_is_val {
-            next_is_val = false;
-            bcgen_op.push_arg(arg);
-            continue;
-        }
 
-        // Skip any arguments explicitly blacklisted
-        if clang_support::is_blacklisted_clang_argument(bc_opts.strict, arg) {
-            skip_next = clang_support::next_arg_is_option_value(arg);
-            continue;
-        }
+        // let next_is_value = clang_support::next_arg_is_option_value(arg);
 
-        if arg.to_str().map_or(false, |s| bc_opts.remove_arguments.is_match(s)) {
-            // Reject arguments matching any of the user-provided regexes.  Note
-            // that this is of course as unsafe as users make it.  In
-            // particular, rejecting '-o' would be very bad.
-            skip_next = clang_support::next_arg_is_option_value(arg);
-            continue;
-        } else {
-            if ! arg.to_string_lossy().starts_with("-o") {
-                if arg.to_string_lossy().starts_with("-") {
-                    bcgen_op.push_arg(arg);
-                    next_is_val = clang_support::next_arg_is_option_value(arg);
-                } else {
-                    bcgen_op.add_input_file(&FileArg::loc(arg));
-                }
+        // Skip any arguments explicitly blacklisted or any matching any of the
+        // user-provided regexes.  Note that this is of course as unsafe as users
+        // make it.  In particular, rejecting '-o' would be very bad.
+        if clang_support::is_blacklisted_clang_argument(bc_opts.strict, arg) ||
+            bc_opts.remove_arguments.is_match(arg.to_string_lossy().to_mut())
```

Comment:
I didn't look in detail, but I was surprised to see this call to `to_mut`. Does `is_match` do something weird and mutable?

Reply:
The `to_string_lossy()` call returns `Cow<'_, str>`, but `is_match` takes a `&str`.  The `Cow` allows `to_mut` and `into_owned`, so the former was preferred; the nice thing about `Cow::to_mut` is that even though that's a mutable reference, the copy is only performed when needed (i.e. here it should not copy).   Since `Cow` implements `Deref`, I could also have done `...is_match(&*arg.to_string_lossy())`, but that seemed like it would be even more confusing to the reader (it looks like a useless no-op).  I can switch to the latter if you prefer, though.